src/main/java/core/basesyntax/service/operations/SupplyOperation.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax.service.operations;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction transaction) {
+        if (Storage.fruits.containsKey(transaction.getFruit())
+                && transaction.getOperation() == FruitTransaction.Operation.SUPPLY) {

```

Comment:
The check for operation type (`transaction.getOperation() == FruitTransaction.Operation.SUPPLY`) seems redundant as this code is within the 'SupplyOperation' class. It's expected that this would only be used for SUPPLY operations.

Reply:
ok, done